src/test/daml/Daml/Finance/Instrument/Bond/Test/Util.daml

```diff
@@ -121,7 +121,27 @@ verifyNoLifecycleEffects readAs today instrument settlers issuer observableCids
   (bondLifecycleCid2, effectCids) <- lifecycleInstrument readAs today instrument settlers issuer observableCids
   assertMsg ("There should be no lifecycle effects on " <> show today) (null effectCids)
 
+-- | Verify the payments from a payment date of a swap (excluding settlement)
+lifecycleAndVerifyBondPaymentEffects : [Party] -> Date -> Instrument.K -> Parties -> Party -> Party -> [(Text, Parties)] -> Party -> [ContractId Observable.I] -> [Instrument.Q] -> [Instrument.Q] -> Script Instrument.K
+lifecycleAndVerifyBondPaymentEffects readAs today instrument settlers issuer investor obs custodian observableCids expectedConsumedQuantities expectedProducedQuantities = do
+  (swapLifecycleCid, [effectCid]) <- lifecycleInstrument readAs today instrument settlers issuer observableCids
+
+  newSwapInstrumentKey <- submit issuer do
+    Instrument.toKey <$> exerciseCmd @Instrument.I (coerceContractId swapLifecycleCid) Instrument.GetView with viewer = issuer
+
+  -- Get the effect
+  effectView <- submitMulti [investor] readAs do
+    exerciseCmd effectCid Effect.GetView with viewer = investor
+
+  -- Verify that the consumed/produced quantities match the expected ones
+  assertMsg "The consumed quantities do not match the expected ones" (sort expectedConsumedQuantities == sort effectView.consumed)
+  assertMsg "The produced quantities do not match the expected ones" (sort expectedProducedQuantities == sort effectView.produced)
+
+  pure newSwapInstrumentKey
+
 -- | Verify a coupon payment of a bond (excluding the last date, which also includes a redemption amount).
+-- Since we have moved to unit tests, this function is mainly kept for the docs.
+-- It contains the code snippets needed for the bond lifecycling tutorial.
```

Comment:
I think it would make sense to remove it from the tests and create a separate runnable tutorial.

Reply:
I have opened a separate issue to review this tutorial, including the code it depends on: https://github.com/digital-asset/daml-finance/issues/215